### DISCUSSION TITLE
Is there support for multiple air purifiers?

### DISCUSSION THREAD

**Discussion by PGSeattle** [RaresAil/homebridge-levoit-air-purifier#73] [Q&A] [answered]
Love the bridge - thank you!

It works wonderfully for 1 of my 2 Levoit Core400S (Firmware 1.3.1), but not both.

For the one that does **not** work, it shows up in the Apple Home app, but when I tap "On", it flashes on for second, but then goes back to off. Same for other controls (like the "Mode" toggle).

So...I'm just wondering...is this expected (that only 1 device works)? Or could this potentially be an entirely different issue?

**Comment by RaresAil** [comment]
Hey, is definitely an issue, can you provide some logs from homebridge?

**Reply by PGSeattle** [reply]
Yes. I added "enableDebugMode": true, and was looking for anything obvious, but didn't see anything that made sense to me. Please let me know if I can help troubleshoot.
[log.txt](https://github.com/RaresAil/homebridge-levoit-air-purifier/files/11995839/log.txt)

**Reply by PGSeattle** [reply]
I wasn't seeing any errors in the logs when I attempted to turn on/off via the Apple Home app or change any settings in the Home app settings.

**Reply by RaresAil** [reply]
Something is not right, i see a lot of “ device reboot command received” what os/device are you using?

**Reply by PGSeattle** [reply]
Sorry for the reboots. I rebooted a few times after changing settings to see if that helped. I'm using an iPhone 14 Pro Max with iOS 16.5.1 (always latest/current public release) for the Apple Home app during the time the logs were logging, but I also use an iPad, or a Mac. My Homebridge version is 1.6.0, Version 4.3.0 (latest), Node 16.20.1, from HOOBS.org, Model HSLF-2.

**Reply by almonde** [reply]
I also see similar behavior with 2 x 200s units.

When I have 1 unit added to my VeSync account I have perfect control of it from HomeBridge and the Home app.  When I added the 2nd I then started having issues where generally control from the Home app was unreliable, either didn't work, or if it did it made the other unit's state in the Home app go 'Turning on' or 'Turning off' and stay that way.

It seems the control from Homebridge UI works but the control from the home app and also automations (Home hubs) does not work reliably with 2 units added.

I plan to add the second one back in day or so to see if just removing from VeSync and re-adding will help.

I also wonder if I can run 2 child bridge instances, with 2 Vesync accounts as a workaround.?

**Reply by almonde** [reply]
Just to update, I removed one unit from Vesync then reset the plugin to flush out the cache. 1 unit was working as expected for 24 hrs so I added the 2nd unit back to vesync and resorted homebridge. Both units appear to be working as expected in the home app now, will monitor.

**Reply by RaresAil** [reply]
It shouldn't be an issue with multiple devices, maybe it can be from HOOBS? I never used the device, i use a RPi 4 with 8 GB of  ram

**Reply by almonde** [reply, marked as answer]
I'm running Homebridge on a Dell optiplex 7060 with 16gb RAM running Mint, no HOOBS. 

I think I might have worked out what causes the issue I was seeing. It seems to occur when I put both purifiers in the same automation, eg: both turn on at 8am at speed 3, then both turn off 2 hours later.   I suspect that making 2 API requests at basically the same time is falling foul of the 500 ms rate limiter mitigation in the plug in?  If I make sure both purifiers live in their own automations, then they seem to be working reliably for over a week.  I could try using Shortcuts and automation to put a sleep 1s step between the purifier requests to see if that cures it, but so far just having their own automations seems to be working.
